### ISSUE TITLE
We need an easy service to get the ascii code point from a literal char

### ISSUE THREAD

**Issue by xymus** [nitlang/nit#1718]
The multiplication of `0x43u8` instead of the good old literal `'C'` makes some already complex code even harder to read, debug and maintain. We need an API that draws attention to the readable ASCII character while making it clear that is is ASCII.

It could look like:

~~~
'C'.ascii # With an assert that the receiver is valid ascii
~~~

Since this problem is mostly on literals, maybe add something to the grammar:

~~~
'C'a # a for ascii (This is probably my favorite)
'C'u8 # for an UTF-8 char on 8 bits with a check when parsing literals
~~~

Or an annotation with a transformation at compilation:

~~~
'C'@ascii -> Byte
~~~

Or a new integer style literal (I don't like it but it's still an improvement):

~~~
0aC # Goes with 0x43u8 and 0b0110_0011u8
~~~

Please @R4PaSs , pretty much any API will do, but no more lines like `tmp >= 0x61u8 and tmp <= 0x66u8` to check if `tmp` is an ascii lower case!

**Comment by R4PaSs** [comment]
I agree on a grammatical rule with bounds checking, you should not be able to write something like `'ま'A`, though I'd think that if something easy is used for ASCII, maybe something similar could be used on literals for which we'd like to get the code point.

I thought of reusing the Int suffixes, though I'm not sure how elegant this is.
`'a'u8 => 0x61u8 (Byte)`
`'ま'u32 => 0x307E (Int)`

The problem with this is what to do with `u16` ? ~~UTF-16 codets ?~~ (HELL NO.)

Maybe re-using the type info ?
`'a'b => 0x61u8 (Byte)`
`'ま'i => 0x307E (Int)`

But this is not clear we want ASCII, maybe we'd want the 8-bit truncated code point (for ISO 8859-1 for example), so I'm not sure we should use either of the two propositions above.

Maybe having a prefix requesting the charset-equivalent of a char if applicable ?

`'a'asc => 0x61u8 (Byte)`
`'ま'uni => 0x307E (Int)`

This is not the worse imho, but maybe there could be better.

> AST transformations are evil
Jean Privat, 2015

This pretty much sums up the defects in the annotation-style rule. Though this could is clear, this is a bit heavy to use imho.

I'm not sure which is best or if anyone has a better idea ?

**Comment by R4PaSs** [comment]
After more thinking. maybe just migrating `ascii` to return `Byte` values for the right table might be enough, the actual `ascii` will be replaced by a `code_point` method with the same implementation as now

**Comment by privat** [comment]
closed by #1726 and #1991
